e107_core/shortcodes/batch/user_shortcodes.php

```diff
@@ -462,11 +462,28 @@ function sc_user_forum_link($parm=null)
 	
 	function sc_user_sendpm($parm=null)
 	{
-		$pref = e107::getPref();
-		$tp = e107::getParser();
+//		$pref = e107::getPref();
+//		why is this orphan $pref here?
+//		$tp = e107::getParser();
 		if(e107::isInstalled("pm") && ($this->var['user_id'] > 0))
 		{
-		  return $tp->parseTemplate("{SENDPM={$this->var['user_id']}}");
+			$parms_str = 'user='.$this->var['user_id'];
+
+			if ($parm) {
+				$parms_str .='&'.implode('&', array_map(
+		            function($k, $v) { 
+                		return $k . '=' . $v;
+            		}, 
+            		array_keys($parm), 
+            		array_values($parm)
+            	)
+        		);
+			}
+
```

Comment:
@rica-carv I believe this can be replaced with http_build_qry(); 

$parm = ['key1' => 'value1', 'key2' => 'value2'];
$query_string = http_build_query($parm);
// Output: key1=value1&key2=value2